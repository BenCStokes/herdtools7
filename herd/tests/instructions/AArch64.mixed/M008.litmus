AArch64 M008
"PodWWPh0 Rfeh0h0 Amo.Cash0h0 PodWWh0P Coe"
Orig=PodWWPh0 Rfeh0h0 Amo.Cash0h0 PodWWh0P Coe
(* Test bug fixed by PR #296 *)
{
0:X1=x; 0:X3=y;
1:X0=y; 1:X4=x;
}
 P0           | P1              ;
 MOV W0,#2    | MOV W1,#257     ;
 STR W0,[X1]  | MOV W2,#514     ;
 MOV W2,#257  | CASH W1,W2,[X0] ;
 STRH W2,[X3] | MOV W3,#1       ;
              | STR W3,[X4]     ;
exists ([x]=0x2 /\ [y]=0x202 /\ 1:X1=0x101)
