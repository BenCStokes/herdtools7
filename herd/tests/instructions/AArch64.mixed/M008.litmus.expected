Test M008 Allowed
States 4
1:X1=0x0; [x]=0x1; [y]=0x101;
1:X1=0x0; [x]=0x2; [y]=0x101;
1:X1=0x101; [x]=0x1; [y]=0x202;
1:X1=0x101; [x]=0x2; [y]=0x202;
Ok
Witnesses
Positive: 1 Negative: 3
Condition exists ([x]=0x2 /\ [y]=0x202 /\ 1:X1=0x101)
Observation M008 Sometimes 1 3
Hash=dce7776cf1382a6e5b1afce5eab68a83

